src/constraints/StringConstraints.ts

```diff
@@ -44,3 +61,93 @@ export function stringLengthNe(length: number): IConstraint<string> {
 	const expected = `expected.length !== ${length}`;
 	return stringLengthComparator(ne, 's.string.lengthNe', expected, length);
 }
+
+export function stringEmail(): IConstraint<string> {
+	return {
+		run(input: string) {
+			return validateEmail(input)
+				? Result.ok(input)
+				: Result.err(new ConstraintError('s.string.email', 'Invalid email address', input, 'expected to be an email address'));
+		}
+	};
+}
+
+export function stringRegexValidator(type: `s.string.${'uuid' | 'regex'}`, expected: string, regex: RegExp): IConstraint<string> {
```

Comment:
This is a private function, just like `stringLengthComparator`.
```suggestion
function stringRegexValidator(type: StringConstraintName, expected: string, regex: RegExp): IConstraint<string> {
```